src/ActionPreviewHeader.jsx

```diff
@@ -1,44 +1,55 @@
-import React from 'react';
+import React, { Component } from 'react';
 
-const ActionPreviewHeader = ({
-  styling, inspectedPath, onInspectPath, tab, onSelectTab
-}) => {
-  return (
-    <div key='previewHeader' {...styling('previewHeader')}>
-      <div {...styling('tabSelector')}>
-        {['Action', 'Diff', 'State'].map(t =>
-          <div onClick={() => onSelectTab(t)}
-               key={t}
-               {...styling([
-                 'selectorButton',
-                 t === tab && 'selectorButtonSelected'
-               ], t === tab)}>
-            {t}
-          </div>
-        )}
-      </div>
-      <div {...styling('inspectedPath')}>
-        {inspectedPath.length ?
-          <span {...styling('inspectedPathKey')}>
-            <a onClick={() => onInspectPath([])}
-               {...styling('inspectedPathKeyLink')}>
-              {tab}
-            </a>
-          </span> : tab
-        }
-        {inspectedPath.map((key, idx) =>
-          idx === inspectedPath.length - 1 ? key :
-          <span key={key}
-             {...styling('inspectedPathKey')}>
-            <a onClick={() => onInspectPath(inspectedPath.slice(0, idx + 1))}
-               {...styling('inspectedPathKeyLink')}>
-              {key}
-            </a>
-          </span>
-        )}
+class ActionPreviewHeader extends Component {
+  constructor(props) {
+    super(props);
+    this.tabs = ['Action', 'Diff', 'State'];
```

Comment:
Is there any reason to store it in `state` instead of `this.tabs` since setting it will not require rerendering the component? I would prefer not to use states here at least to indicate that this component is stateless.